lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,359 @@
+                        create_energy_dispersion_hdu,
+                        create_background_2d_hdu)
+from pyirf.irf import (effective_area_per_energy,
+                        energy_dispersion,
+                        effective_area_per_energy_and_fov,
+                        background_2d)
+from pyirf.spectral import (
+    calculate_event_weights,
+    PowerLaw,
+    CRAB_HEGRA,
+    IRFDOC_PROTON_SPECTRUM,
+    IRFDOC_ELECTRON_SPECTRUM,
+)
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+from pyirf.binning import create_bins_per_decade #, add_overflow_bins
+
+__all__ = [
+    'IRFFITSWriter'
+    ]
+
+class IRFFITSWriter(Tool):
+    name = "IRFFITSWriter"
+    description = __doc__
+
+    input_gamma_dl2 = traits.Path(
+        help="Input MC gamma DL2 file",
+        exists=True,
+        directory_ok=False,
+        file_ok=True
+        ).tag(config=True)
+
+    input_proton_dl2 = traits.Path(
+        help="Input MC proton DL2 file",
+        exists=True,
+        directory_ok=False,
+        file_ok=True
+        ).tag(config=True)
+
+    input_electron_dl2 = traits.Path(
+        help="Input MC electron DL2 file",
+        exists=True,
+        directory_ok=False,
+        file_ok=True
+        ).tag(config=True)
+
+    output_irf_file = traits.Path(
+        help="IRF output file",
+        directory_ok=False,
+        file_ok=True,
+        default_value="./irf.fits.gz"
+        ).tag(config=True)
+
+    point_like = traits.Bool(
+        help="True for point-like IRF, False for Full Enclosure",
+        default_value=False,
+        ).tag(config=True)
+
+    config_file = traits.Path(
+        help="Config file for selection cuts",
+        directory_ok=False,
+        file_ok=True,
+        ).tag(config=True)
+
+    aliases = {
+        "input_gamma_dl2" : "IRFFITSWriter.input_gamma_dl2",
+        "fg" : "IRFFITSWriter.input_gamma_dl2",
+        "input_proton_dl2" : "IRFFITSWriter.input_proton_dl2",
+        "fp" : "IRFFITSWriter.input_proton_dl2",
+        "input_electron_dl2" : "IRFFITSWriter.input_electron_dl2",
+        "fe" : "IRFFITSWriter.input_electron_dl2",
+        "output_irf_file" : "IRFFITSWriter.output_irf_file",
+        "o" : "IRFFITSWriter.output_irf_file",
+        "point_like" : "IRFFITSWriter.point_like",
+        "pnt" : "IRFFITSWriter.point_like",
+        "config_file" : "IRFFITSWriter.config_file",
+        "conf" : "IRFFITSWriter.config_file",
+        }
+
+    flag = {
+        "point_like": (
+            {"IRFFITSWriter": {
+                        "point_like": False}
+                        },
+            "Full Enclosure IRFs will be produced"
+            ),
+        }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.mc_particle = None
+        self.cuts = None
+        self.hdus = None
+        self.effective_area = None
+        self.edisp = None
+        self.backgroud = None
+        # self.psf = None
```

Comment:
why is the PSF part commented out?

Reply:
I have commented out PSF in this PR, as the Tools are currently used mainly for producing point-like IRFs, with the option to later complete it with full enclosure IRFs as well when the need arises.